examples/todolist/tests/bindings/test_todolist.kts

```diff
@@ -50,6 +50,29 @@ todo.addItems(listOf("bobo", "fofo"))
 assert(todo.getItems().size == 9)
 assert(todo.getItems()[7] == "bobo")
 
+assert(getDefaultList() == null)
+TodoList().use { todo2 ->
+    setDefaultList(todo)
+    getDefaultList()!!.use { default ->
+        assert(todo.getEntries() == default.getEntries())
+        assert(todo2.getEntries() != default.getEntries())
+    }
+
+    todo2.makeDefault()
+    getDefaultList()!!.use { default ->
+        assert(todo.getEntries() != default.getEntries())
+        assert(todo2.getEntries() == default.getEntries())
+    }
+
+    todo.addItem("Test liveness after being demoted from default")
+    assert(todo.getLast() == "Test liveness after being demoted from default")
+
+    todo2.addItem("Test shared state through local vs default reference")
+    getDefaultList()!!.use { default ->
+        assert(default.getLast() == "Test shared state through local vs default reference")
+    }
+}
```

Comment:
This test shows a bit of the awkwardness of working with object references in Kotlin, where we have to make sure each instance is explicitly destroyed when we're finished with it. It's manageable, but easy to forget about. I'd like to see us take a fresh look at e.g. weak references for doing this more automatically, but that's well outside the scope of this PR.

Reply:
It might be worth summarizing this comment in the example - a casual reader might not realize the importance of the pattern used.